datus/tools/func_tool/context_search.py

```
@@ -96,7 +96,7 @@ def list_subject_tree(self) -> FuncToolResult:
         except ValueError as exc:
             return FuncToolResult(success=0, error=str(exc))
         except Exception as exc:  # pragma: no cover - defensive logging
-            logger.error("Failed to assemble domain taxonomy: %s", exc)
+            logger.error("Failed to assemble domain taxonomy: {exc}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)